test/integration/api/authentication.spec.js

```diff
@@ -49,8 +112,122 @@ const authorizationServerInfo = {
 let apiServerUrl
 let publicPlatformId
 
+async function checkOAuthProcess ({ t, provider, ssoConnection, afterAuthenticationUrl, executeBrowserScenario }) {
+  const systemKey = getSystemKey()
+
+  // add SSO connection
+  await request(t.context.serverUrl)
+    .patch('/config/private')
+    .set({
+      'x-platform-id': t.context.platformId,
+      'x-stelace-system-key': systemKey,
+      'x-stelace-env': t.context.env
+    })
+    .send({
+      stelace: {
+        ssoConnections: {
+          [provider]: Object.assign({}, ssoConnection, {
+            protocol: 'oauth2',
+            afterAuthenticationUrl,
+            active: true
+          })
+        }
+      }
+    })
+    .expect(200)
+
+  const browser = await puppeteer.launch({
+    executablePath: '/usr/bin/chromium-browser',
+    args: ['--disable-dev-shm-usage', '--no-sandbox']
+  })
+
+  const page = await browser.newPage()
+
+  // Chrome on Linux
+  await page.setUserAgent('Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36')
+
+  await executeBrowserScenario({ browser, page })
+
+  await new Promise(resolve => setTimeout(resolve, 1000))
```

Comment:
It sounds like a fragile and magic delay. You need to add some short explanation _at least_ in case test breaks because of this.

Reply:
Yes thanks, I'll add an explanation.